pkg/pop/poptoken.go

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package pop
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"log"
+	"math/big"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+const popTokenType = "pop"
+
+type PopAuthenticationScheme struct {
```

Comment:
Eveything about the authnscheme from msal could be moved to a dedicated file in this same pkg `authnscheme.go`. 

`poptoken.go` should only be the interface and the implementation of it

Reply:
Moved `PoPAuthenticationScheme` code to `authnscheme.go`